### PR TITLE
Migrate to Java 17

### PR DESCRIPTION
**Pull request by gabilang**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: .github/workflows/build-timestamped-master.yml [hunk 1]
```diff
@@ -10,12 +10,12 @@ jobs:
     runs-on: ubuntu-latest
     if: github.repository_owner == 'ballerina-platform'
     steps:
-      - uses: actions/checkout@v2
-      - name: Set up JDK 11.0.7
-        uses: actions/setup-java@v2
+      - uses: actions/checkout@v3
+      - name: Set up JDK 17.0.7
```
Comment (warunalakshitha) [inline .github/workflows/build-timestamped-master.yml:14]:
```suggestion
      - name: Set up JDK 17
```

File: .github/workflows/publish-release.yaml [hunk 2]
```diff
@@ -7,12 +7,12 @@ jobs:
   publish-release:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - name: Set up JDK 11.0.7
-        uses: actions/setup-java@v2
+      - uses: actions/checkout@v3
+      - name: Set up JDK 17.0.7
```
Comment (warunalakshitha) [inline .github/workflows/publish-release.yaml:11]:
```suggestion
      - name: Set up JDK 17
```

File: .github/workflows/pull-request.yml [hunk 3]
```diff
@@ -34,12 +34,12 @@ jobs:
     runs-on: windows-latest
 
     steps:
-      - uses: actions/checkout@v2
-      - name: Set up JDK 11.0.7
-        uses: actions/setup-java@v2
+      - uses: actions/checkout@v3
+      - name: Set up JDK 17.0.7
```
Comment (warunalakshitha) [inline .github/workflows/pull-request.yml:38]:
```suggestion
      - name: Set up JDK 17
```

File: graphql-cli/src/main/java/io/ballerina/graphql/cmd/Utils.java [hunk 4]
```diff
@@ -65,7 +66,7 @@ public class Utils {
      * @return               the `GraphQLSchema` instance
      */
     public static Constructor getProcessedConstructor() {
-        Constructor constructor = new Constructor(Config.class);
+        Constructor constructor = new Constructor(Config.class, new LoaderOptions());
```
Comment (warunalakshitha) [inline graphql-cli/src/main/java/io/ballerina/graphql/cmd/Utils.java:69]: Why we need this change?

Comment (gabilang) [reply graphql-cli/src/main/java/io/ballerina/graphql/cmd/Utils.java:69]: The `SnakeYaml` version `1.33` has vulnerabilities and with Java-17 the version `2.0` has been suggested to use. The version `2.0` provides the constructor which requires a parameter for load operations configuration.

**Review by warunalakshitha — APPROVED** [review]

**Pull request merged**